monkey/monkey_island/cc/server_utils/file_utils.py

```diff
@@ -0,0 +1,59 @@
+import os
+
+from monkey_island.cc.environment.utils import is_windows_os
+
+
+def expand_path(path: str) -> str:
+    return os.path.expandvars(os.path.expanduser(path))
+
+
+def has_expected_permissions(path: str, expected_permissions: int) -> bool:
+    if is_windows_os():
+        return _has_expected_windows_permissions(path, expected_permissions)
+
+    return _has_expected_linux_permissions(path, expected_permissions)
+
+
+def _has_expected_linux_permissions(path: str, expected_permissions: int) -> bool:
+    file_mode = os.stat(path).st_mode
+    file_permissions = file_mode & 0o777
+
+    return file_permissions == expected_permissions
+
+
+def _has_expected_windows_permissions(path: str, expected_permissions: int) -> bool:
+    # checks that user has `expected_permissions` and everyone else has no permissions
```

Comment:
@shreyamalviya This comment is still misleading, as we don't check admin permissions at all, and admin is included in "everyone else".